saito-core/src/core/data/network.rs

```diff
@@ -67,8 +82,165 @@ impl Network {
+            peer.initiate_handshake(&self.io_interface, wallet.clone(), configs.clone())
+                .await
+                .unwrap();
+        }
+
+        peers.index_to_peers.insert(peer_index, peer);
+        info!("new peer added : {:?}", peer_index);
+    }
+    pub async fn handle_handshake_challenge(
+        &self,
+        peer_index: u64,
+        challenge: HandshakeChallenge,
+        wallet: Arc<RwLock<Wallet>>,
+        configs: Arc<RwLock<Configuration>>,
+    ) {
+        let mut peers = self.peers.write().await;
+        let peer = peers.index_to_peers.get_mut(&peer_index);
+        if peer.is_none() {
+            todo!()
+        }
+        let peer = peer.unwrap();
+        peer.handle_handshake_challenge(
+            challenge,
+            &self.io_interface,
+            wallet.clone(),
+            configs.clone(),
+        )
+        .await
+        .unwrap();
+    }
+    pub async fn handle_handshake_response(
+        &self,
+        peer_index: u64,
+        response: HandshakeResponse,
+        wallet: Arc<RwLock<Wallet>>,
+        blockchain: Arc<RwLock<Blockchain>>,
+    ) {
+        debug!("received handshake response");
+        let mut peers = self.peers.write().await;
+        let peer = peers.index_to_peers.get_mut(&peer_index);
+        if peer.is_none() {
+            todo!()
+        }
+        let peer = peer.unwrap();
+        peer.handle_handshake_response(response, &self.io_interface, wallet.clone())
+            .await
+            .unwrap();
+        if peer.handshake_done {
+            debug!(
+                "peer : {:?} handshake successful for peer : {:?}",
+                peer.peer_index,
+                hex::encode(peer.peer_public_key)
+            );
+            // start block syncing here
+            self.request_blockchain_from_peer(peer_index, blockchain.clone())
+                .await;
+        }
+    }
+    pub async fn handle_handshake_completion(
+        &self,
+        peer_index: u64,
+        response: HandshakeCompletion,
+        blockchain: Arc<RwLock<Blockchain>>,
+    ) {
+        debug!("received handshake completion");
+        let mut peers = self.peers.write().await;
+        let peer = peers.index_to_peers.get_mut(&peer_index);
+        if peer.is_none() {
+            todo!()
+        }
+        let peer = peer.unwrap();
+        let result = peer
+            .handle_handshake_completion(response, &self.io_interface)
+            .await;
+        if peer.handshake_done {
+            debug!(
+                "peer : {:?} handshake successful for peer : {:?}",
+                peer.peer_index,
+                hex::encode(peer.peer_public_key)
+            );
+            // start block syncing here
+            self.request_blockchain_from_peer(peer_index, blockchain.clone())
+                .await;
+        }
+    }
+    async fn request_blockchain_from_peer(
```

Comment:
If this method is moved into the network scope, why not move the other two related methods (which will be invoked due to this)

1. process_incoming_blockchain_request
2. process_incoming_block_hash

into the same scope just to increase the clarity?

Reply:
Then shall we move everything into a request_processor, whatever the requests coming from the peers will be handled by the peer request processor